examples/common/mod.rs

```diff
@@ -73,7 +99,98 @@ impl Default for PacketStats {
         PacketStats {
             start_time: std::time::Instant::now(),
             current: 0,
-            history: heapless::HistoryBuffer::new(),
+            bytes: 0,
+            message_history: History::new(),
+            byte_history: History::new(),
+        }
+    }
+}
+
+fn log_metric(metric_name: &str, metric_value: u128, elapsed_micros: u128, history: &History) {
+    log::info!(
+        "{}: {:>8} | 1s average: {:>8}/s) | {}s average: {:>8}/s{}",
+        metric_name,
+        metric_value,
+        metric_value * 1_000_000 / elapsed_micros,
+        PACKET_STATS_WINDOW_SIZE,
+        history.as_slice().iter().sum::<u128>() * 1_000_000
+            / (history.len() as u128)
+            / elapsed_micros,
+        if history.len() < PACKET_STATS_WINDOW_SIZE {
+            " (?)"
+        } else {
+            ""
+        },
+    );
+}
+
+fn log_sys_metrics() {
+    let sys = System::new();
+
+    match sys.networks() {
+        Ok(netifs) => {
+            log::info!("Networks:");
+            for netif in netifs.values() {
+                log::info!("{} ({:?})", netif.name, netif.addrs);
+            }
+        }
+        Err(x) => log::info!("Networks: error: {}", x),
+    }
+
+    match sys.networks() {
+        Ok(netifs) => {
+            log::info!("Network interface statistics:");
+            for netif in netifs.values() {
+                log::info!(
+                    "{} statistics: ({:?})",
+                    netif.name,
+                    sys.network_stats(&netif.name)
+                );
+            }
         }
+        Err(x) => log::info!("Networks: error: {}", x),
+    }
+
+    match sys.memory() {
+        Ok(mem) => log::info!(
+            "Memory: {} used / {} ({} bytes) total ({:?})",
+            saturating_sub_bytes(mem.total, mem.free),
+            mem.total,
+            mem.total.as_u64(),
+            mem.platform_memory
+        ),
+        Err(x) => log::info!("Memory: error: {}", x),
+    }
+
+    match sys.load_average() {
+        Ok(loadavg) => log::info!(
+            "Load average: {} {} {}",
+            loadavg.one,
+            loadavg.five,
+            loadavg.fifteen
+        ),
+        Err(x) => log::info!("Load average: error: {}", x),
+    }
+
+    match sys.cpu_load_aggregate() {
+        Ok(cpu) => {
+            log::info!("Measuring CPU load...");
+            thread::sleep(Duration::from_secs(1));
```

Comment:
Why?

Reply:
Wasn't sure what all metrics we want, but maybe it is worthwhile having more metrics captured here? As per your other comment, I agree maybe not in hot path, but somewhere else? This way we could have a single tool to use, but I am also okay with removing if you feel it is too much trouble :)